src/pathfinder/water_regions.cpp

```diff
@@ -342,38 +346,19 @@ void VisitWaterRegionPatchNeighbors(const WaterRegionPatchDesc &water_region_pat
 	}
 }
 
-std::vector<WaterRegionSaveLoadInfo> GetWaterRegionSaveLoadInfo()
-{
-	std::vector<WaterRegionSaveLoadInfo> result;
-	for (WaterRegion &region : _water_regions) result.push_back({ region.IsInitialized() });
-	return result;
-}
-
-void LoadWaterRegions(const std::vector<WaterRegionSaveLoadInfo> &save_load_info)
-{
-	_water_regions.clear();
-	_water_regions.reserve(save_load_info.size());
-	TWaterRegionIndex index = 0;
-	for (const auto &loaded_region_info : save_load_info) {
-		const int region_x = index % GetWaterRegionMapSizeX();
-		const int region_y = index / GetWaterRegionMapSizeX();
-		WaterRegion &region = _water_regions.emplace_back(region_x, region_y);
-		if (loaded_region_info.initialized) region.ForceUpdate();
-		index++;
-	}
-}
-
 /**
- * Initializes all water regions. All water tiles will be scanned and interconnected water patches within regions will be identified.
+ * Allocates the appropriate amount of water regions for the current map size
  */
-void InitializeWaterRegions()
+void AllocateWaterRegions()
 {
 	_water_regions.clear();
 	_water_regions.reserve(static_cast<size_t>(GetWaterRegionMapSizeX()) * GetWaterRegionMapSizeY());
 
+	Debug(map, 1, "Allocating {} x {} water regions", GetWaterRegionMapSizeX(), GetWaterRegionMapSizeY());
```

Comment:
Is this actually meant to go in the code? If so, please increase the debugging level; doesn't sound worthy of 1 :)

Reply:
Map::allocate had debug level 1 so I used that one, although you could argue that the map is quite a bit more important. I've changed this debug level to 2 and the others to 3.